server_config.go

```diff
@@ -108,6 +109,87 @@ func GenerateAuthKey(username, realm, password string) []byte {
 	return h.Sum(nil)
 }
 
+// EventHandlers is a set of callbacks that the server will call at certain hook points during an
+// allocation's lifecycle. All events are reported with the context that identifies the allocation
+// triggering the event (source and destination address, protocol, username and realm used for
+// authenticating the allocation). It is OK to handle only a subset of the callbacks.
+type EventHandlers struct {
```

Comment:
Should this be called `Callbacks` rather than `EventHandlers`?

Also, should we call it `ServerCallbacks`?  It's not going to appear in a lot of code, perhaps we should be more explicit?

Reply:
Since this is top-level API I guess we'll always use this as `turn.XXX`  from the caller site. This suggests that maybe `ServerCallbacks` would be the right name, WDYT?